packages/core/data-transfer/lib/providers/local-strapi-source-provider/__tests__/test-utils.ts

```diff
@@ -0,0 +1,57 @@
+import { Readable } from 'stream';
+
+/**
+ * Collect every entity in a Readable stream
+ */
+export const collect = <T = unknown>(stream: Readable): Promise<T[]> => {
+  const chunks: T[] = [];
+
+  return new Promise((resolve) => {
+    stream.on('data', (chunk) => chunks.push(chunk));
+    stream.on('end', () => {
+      stream.destroy();
+      resolve(chunks);
+    });
+  });
+};
+
+/**
+ * Create a "Strapi" like object factory based on the
+ * given params and cast it to the correct type
+ */
+export const getStrapiFactory =
```

Comment:
it would be useful to have this in a lot of other circumstances, is there a more general testing utils (outside of data-transfer) where that this would make sense?

Reply:
Of course we would lose the typescript which is the main point, so maybe not for now

Reply:
Yeah, in most of the other tests, we're just defining Strapi as an object, so I'm not sure we need to add a function call. As you said, the benefit here is having a typed constructor.

Also, this getStrapiFactory is meant to be used for the Strapi providers so that it maps to the required property in their constructor